soh/src/overlays/gamestates/ovl_select/z_select.c

```diff
@@ -1101,82 +1156,159 @@ void Select_PrintLoadingMessage(SelectContext* this, GfxPrint* printer) {
     GfxPrint_SetPos(printer, 10, 15);
     GfxPrint_SetColor(printer, 255, 255, 255, 255);
     randomMsg = Rand_ZeroOne() * ARRAY_COUNT(sLoadingMessages);
-    GfxPrint_Printf(printer, "%s", sLoadingMessages[randomMsg]);
+    if (CVar_GetS32("gDebugWarpScreenTranslation", 0)) {
+        switch (gSaveContext.language) {
+            case LANGUAGE_ENG:
+            default:
+                GfxPrint_Printf(printer, "%s", sLoadingMessages[randomMsg].englishMessage);
+                break;
+            case LANGUAGE_GER:
+                GfxPrint_Printf(printer, "%s", sLoadingMessages[randomMsg].germanMessage);
+                break;
+            case LANGUAGE_FRA:
+                GfxPrint_Printf(printer, "%s", sLoadingMessages[randomMsg].frenchMessage);
+                break;
+        }
+    } else {
+        GfxPrint_Printf(printer, "%s", sLoadingMessages[randomMsg].japaneseMessage);
+    }
 }
 
-static const char* sAgeLabels[] = {
-    "17(Adult)",
-    "5(Young)",
```

Comment:
It's fine, it refers to the LINK_AGE_IN_YEARS value which is YEARS_CHILD (5) when child though...